tests/integration/telemetry/stackdriver/stackdriver_filter_audit_test.go

```diff
@@ -0,0 +1,150 @@
+	"strings"
+	"testing"
+	"time"
+
+	"golang.org/x/sync/errgroup"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/scopes"
+	"istio.io/istio/pkg/test/util/file"
+	"istio.io/istio/pkg/test/util/retry"
+	"istio.io/istio/pkg/test/util/tmpl"
+	"istio.io/istio/tests/integration/telemetry"
+)
+
+const (
+	serverAuditAllLogEntry  = "testdata/security_authz_audit/server_audit_all_log.json.tmpl"
+	serverAuditFooLogEntry  = "testdata/security_authz_audit/server_audit_foo_log.json.tmpl"
+	serverAuditBarLogEntry  = "testdata/security_authz_audit/server_audit_bar_log.json.tmpl"
+	serverAuditNoneLogEntry = "testdata/security_authz_audit/server_audit_none_log.json.tmpl"
+	auditPolicyForLogEntry  = "testdata/security_authz_audit/v1beta1-audit-authorization-policy.yaml.tmpl"
+)
+
+// TestStackdriverAuditLogging testing Authz Policy can config stackdriver with audit policy
+func TestStackdriverHTTPAuditLogging(t *testing.T) {
+	framework.NewTest(t).
+		Features("observability.telemetry.stackdriver").
+		Run(func(ctx framework.TestContext) {
+			g, _ := errgroup.WithContext(context.Background())
+
+			ns := getEchoNamespaceInstance().Name()
+			args := map[string]string{
+				"Namespace": ns,
+			}
+			policies := tmpl.EvaluateAllOrFail(t, args, file.AsStringOrFail(t, auditPolicyForLogEntry))
+			ctx.Config().ApplyYAMLOrFail(t, ns, policies...)
+			defer ctx.Config().DeleteYAMLOrFail(t, ns, policies...)
+			t.Logf("Audit policy deployed to namespace %v", ns)
+
+			t.Logf("Traffic sent to namespace %v", ns)
+			for _, cltInstance := range clt {
+				scopes.Framework.Infof("Validating Audit policy and Telemetry for Cluster %v", cltInstance.Config().Cluster.Name())
+				g.Go(func() error {
+					err := retry.UntilSuccess(func() error {
+						if err := sendTrafficForAudit(t, cltInstance); err != nil {
+							return err
+						}
+
+						clName := cltInstance.Config().Cluster.Name()
+						t.Logf("Collect Audit Log for cluster %v", clName)
+
+						errAuditFoo := validateLogs(t, serverAuditFooLogEntry, clName)
+						if errAuditFoo == nil {
+							t.Logf("Foo Audit Log validated for cluster %v", clName)
+						}
+						errAuditBar := validateLogs(t, serverAuditBarLogEntry, clName)
+						if errAuditBar == nil {
+							t.Logf("Bar Audit Log validated for cluster %v", clName)
+						}
+						errAuditAll := validateLogs(t, serverAuditAllLogEntry, clName)
+						if errAuditAll == nil {
+							t.Logf("All Audit Log validated for cluster %v", clName)
+						}
+						errAuditNone := validateLogs(t, serverAuditNoneLogEntry, clName)
+						if errAuditNone == nil {
+							t.Logf("None Audit Log validated for cluster %v", clName)
+						}
+						if errAuditFoo == nil && errAuditBar == nil && errAuditAll == nil && errAuditNone != nil {
+							return nil
+						}
+
+						var errs []string
+						errs = append(errs, errAuditAll.Error(), errAuditFoo.Error(), errAuditBar.Error(), errAuditNone.Error())
```

Comment:
You can create the `errs` before the first validateLogs() and append to it if the error is not nil. Then you can just check len(errs) != 0 in the end instead of specifying errAuditAll, errAuditFoo, errAuditBar and errAuditNone separately.

Reply:
Sure! Thanks!